src/com/baloise/automagic/magic/internal/MagicImpl.groovy

```diff
@@ -0,0 +1,142 @@
+package com.baloise.automagic.magic.internal
+
+import com.baloise.automagic.common.Automagic
+import com.baloise.automagic.common.Registered
+import com.baloise.automagic.credentials.CredentialsService
+import com.baloise.automagic.magic.MagicService
+import com.baloise.automagic.oim.OneItMarketplaceService
+import com.baloise.automagic.properties.PropertyStoreService
+
+class MagicImpl extends Registered implements MagicService {
+
+	
+	PropertyStoreService getProps() { registry.getService(PropertyStoreService) }
+	OneItMarketplaceService getOim() { registry.getService(OneItMarketplaceService)}
+	
+	@Override
+	def magic() {
+		echo "using automagic v"+ Automagic.VERSION
+		
+		if(env.NODE_NAME) error """Automagic is a potentially long running pipeline. 
+		You must not run inside a node but are running on '${env.NODE_NAME}'. 
+		For declarative pipelines please use 'agent none'. 
+		For scripted pipelines make sure you are not running inside a 'node(){}'."""
+		
+		// we do the following in two loops to avoid sleeping inside the node
+		Map<String, String> yamls
+		node(''){
+			checkout scm
+			yamls = findFiles(glob: '.automagic/**/*.yaml').collectEntries{["${it.path}": readYaml(file: it.path)]}
+		}
+		yamls.each{ name,yaml ->
+			doMagic(name, yaml)
+		}
+	}
+	
+	def doMagic(name, yaml) {
+		echo "applying $name" 
+		String ip
+				
+		String ComputerName = props.get('ComputerName-'+yaml.spec.id)
+		if(ComputerName) echo "ComputerName is $ComputerName"
+	
+		def json = oim.getVMDetails(ComputerName)
+		ip = json.Status == 'Success' ? json.Result[0].PrimaryIP : ''
+		if(ip) echo "IP is $ip"
+		
+		String jiraTask = props.get('JIRA-Decommission-ID-'+ComputerName)
+		
+		if(yaml.spec.status == "decommissioned") {
+			echo("Starting Decommissioning procedure")
+			assert ComputerName
+			if(jiraTask) {
+				error "JIRA issue for decommissioning already exists: " + jiraTask
```

Comment:
Do we need another status here in order to avoid generating errors when a yaml gets re-processed but the machine is already decommisioned? (decomission vs decomissioned.) 
If I read the code correctly this would only work properly the first time the yaml is read and the machine has to be decomissioned. Called again it would throw an error. 
Later runs should silently noop/idle if a machine marked for decommission is already decomissioned, or not?